### ISSUE TITLE
Transfer Learning using pretrained weights

### ISSUE THREAD

**Issue by surajmahangade** [facebookresearch/TimeSformer#8]
Hello, I want to do transfer learning using the timeSformer model.
What I want to achieve is to predict 6 outputs(regression) for every 8 frames.
I am a little confused about how exactly to do it.
Any help is appreciated.

Thank you

**Comment by gberta** [comment]
You would need to modify the number of outputs to 6 * 8 in the MODEL.NUM_CLASSES entry of a configuration file. Furthermore, you would need to use L2 loss instead of cross-entropy loss.

**Comment by surajmahangade** [comment]
Thank you for a quick response.
Where should I keep my 6 output values?

**Comment by varad-more** [comment]
Even I've similar doubt any good with this please update here.

**Comment by gberta** [comment]
I'm not sure I understand the issue here. In order for this to work, you simply need to change the number of outputs, and the loss function. This should be pretty trivial to do inside the code.

**Comment by surajmahangade** [comment]
My question is what will change in the dataset preparation. Since I will have 8 outputs.

**Comment by gberta** [comment]
You can create cvs lists of the form "video_path label1,label2,label3,label4,label5,label6". Afterwards, you would need to write a new dataset class, and add it to lib/dataset folder. In this new dataset class, you would read in all 6 of these label values, and then return them in the __getitem__ method (instead of returning a single label). Please take a look at lib/dataset/kinetics.py for a reference on how data loading is done.

**Comment by tcapelle** [comment]
would you mind adding a notebooks loading a pretrained model?

**Comment by surajmahangade** [comment]
Thank you, will try it.
I have one doubt. I changed the number of frames to 2 and loaded one of the models pre-trained weights(to be specific [model](https://www.dropbox.com/s/g5t24we9gl5yk88/TimeSformer_divST_8x32_224_K400.pyth?dl=0). Since the original model is trained on 8 frames, will there be any problem in performance or training I should be aware of?

Thank you.

**Comment by gberta** [comment]
Yes, there might be some generalizations issues if you only use 2 frames with a 8-frame model. However, if you finetune the model on your own data, I believe that it should be fine (as long as your dataset is of reasonable size).

**Comment by surajmahangade** [comment]
Okay, will keep that in mind.
![image](https://user-images.githubusercontent.com/42811380/116736842-11cf6280-aa0e-11eb-998e-3963cf484ecd.png)
Is this normal? While loading pre-trained weights?
I changed back to 8 frames still this is printed.
`Network weights model.head.weight not loaded.
Network weights model.head.bias not loaded.`

Thank you

**Comment by gberta** [comment]
Yes, that's fine. When loading model parameters this operation skips the parameters of the last layer since they don't match the dimensions of your new outputs.
